gen/src/write.rs

```diff
@@ -1649,6 +1649,11 @@ fn write_kj_own(out: &mut OutFile, key: NamedImplKey) {
         "static_assert(alignof(::kj::Own<{}>) == sizeof(void *), \"Static disposers for Own are not supported in workerd-cxx\");",
         inner,
     );
+    writeln!(
+        out,
+        "static_assert(!::std::is_base_of<::kj::Refcounted, {}>::value, \"Value must not inherit from kj::Refcounted\");",
```

Comment:
@LordGoatius, you could use `kj::_::IsRefcounted<{}>` to catch both `Refcounted` and `AtomicRefcounted`: https://github.com/capnproto/capnproto/blob/1a7181a4aae3dea769dbae65bcc00da340d562c8/c%2B%2B/src/kj/memory.h#L92